File: java/src/strategy/SimUDuckApp/Revision4/RedheadDuck.java
package strategy.SimUDuckApp.Revision4;

public class RedheadDuck extends Duck {
	
	public RedheadDuck() {
		setFlyBehavior(new FlyWithWings());
		setQuackBehavior(new Quack());
	}
	
	public void display() {
		System.out.println("I'm a real Red Headed duck");
	}

}

File: src/strategy/SimUDuckApp/Revision4/RedheadDuck.kt
package strategy.SimUDuckApp.Revision4

class RedheadDuck : Duck() {
    init {
        flyBehavior = FlyWithWings()
        quackBehavior = Quack()
    }

    override fun display() {
        println("I'm a real Red Headed duck")
    }
}
